backend/middleware/timeouts/timeouts.go

```diff
@@ -57,18 +61,47 @@ func (m *mid) getDuration(service, method string) time.Duration {
 	return m.defaultTimeout
 }
 
+type unaryHandlerReturn struct {
+	resp interface{}
+	err  error
+}
+
 func (m *mid) UnaryInterceptor() grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		service, method, ok := middleware.SplitFullMethod(info.FullMethod)
 		if !ok {
 			m.logger.Warn("could not parse gRPC method", zap.String("fullMethod", info.FullMethod))
 		}
 
+		// Compute timeout and set-up a context with timeout.
 		timeout := m.getDuration(service, method)
 		ctx, cancel := context.WithTimeout(ctx, timeout)
 		defer cancel()
 
-		return handler(ctx, req)
+		// Create a return channel for the goroutine.
+		resultChan := make(chan unaryHandlerReturn)
+		defer close(resultChan)
+
+		// Spawn the handler in a goroutine so we can return early on timeout if it doesn't complete.
+		go func() {
+			resp, err := handler(ctx, req)
+			select {
+			case <-ctx.Done():
+				m.logger.Warn("handler completed after timeout", zap.String("service", service), zap.String("method", method))
```

Comment:
well the handler has to complete to ever get to this point. i.e. if is respecting the context it would return err on timeout.

still i think you're on to something, and maybe this is what you were getting at. if the handler respected `Done()` it would be better to return the handler's error than logging this error.

will think on it a bit more and hopefully get a test up for these cases.

Reply:
Ah yes true I suppose it must complete to get to this point. I was a bit confused around the wording of the warning message which is why I asked. 

But if err is not nil here then we can also log it and upgrade this warn to an error perhaps.

Reply:
fixed, now using a done channel to track the overall execution of the middleware, which would return after the boost has elapsed, allowing the handler to return its err within the boost period.